KPI-Schedule/Controller/ScheduleController.swift

```diff
@@ -8,24 +8,134 @@ class ScheduleController: UITableViewController {
         super.viewDidLoad()
-
+        scheduleManager.delegate = self
+        navigationItem.title = defaults.string(forKey: "selectedGroupName") ?? "ІМ-13"
+        scheduleManager.getSchedule(id: defaults.string(forKey: "selectedGroupId") ?? defaultId)
 
     }
     
     @IBAction func changeWeek(_ sender: UISegmentedControl) {
-    let selectedSchedule = sender.selectedSegmentIndex == 1 ? secondWeekSchedule : firstWeekSchedule
-    scheduleWeek = selectedSchedule
+        let selectedSchedule = sender.selectedSegmentIndex == 1 ? secondWeekSchedule : firstWeekSchedule
+        scheduleWeek = selectedSchedule
     
-    DispatchQueue.main.async {
+        DispatchQueue.main.async {
         self.tableView.reloadData()
+        }
+    }
+    
+    @IBAction func unwindToThisViewController(segue: UIStoryboardSegue) {
+        navigationItem.title = defaults.string(forKey: "selectedGroupName") ?? "ІМ-01"
+        scheduleManager.getSchedule(id: defaults.string(forKey: "selectedGroupId") ?? defaultId)
     }
 }
 
     
-    
 
+extension ScheduleController {
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        daysWithPairs = []
+        for day in scheduleWeek {
+            if !day.pair.isEmpty {
+                daysWithPairs.append(day)
+            }
+        }
+
+        return daysWithPairs.count
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return daysWithPairs[section].pair.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "PairCell", for: indexPath)
+        
+        var configuration = cell.defaultContentConfiguration()
+        let pair = daysWithPairs[indexPath.section].pair[indexPath.row]
+        let pairName = pair.type != "" ? "\(pair.name) (\(pair.type))" : pair.name
+        configuration.text = pairName
+        configuration.secondaryText = pair.time
+        configuration.secondaryTextProperties.color = .systemBlue
+        cell.contentConfiguration = configuration
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let pairs = daysWithPairs[indexPath.section].pair[indexPath.row]
+        
+        let alertTitle = pairs.type != "" ? "\(pairs.name) (\(pairs.type))" : pairs.name
+        lazy var alertMessage: String = {
+            if pairs.teacherName != "" {
+                if pairs.place != "" {
+                    return "\(pairs.teacherName)\n\(pairs.place)"
+                } else {
+                    return "\(pairs.teacherName)"
+                }
+            } else {
+                if pairs.place != "" {
+                    return "\(pairs.place)"
+                } else {
+                    return ""
+                }
+            }
+        }()
```

Comment:
Rework for early return and remove inverted operands, It is not easier to read. Also, if you can, use method as .empty or similar